pkg/machine/config.go

```diff
@@ -493,3 +524,76 @@ func WaitAndPingAPI(sock string) {
 		logrus.Warn("API socket failed ping test")
 	}
 }
+
+func (dl Download) NewFcosDownloader(imageStream FCOSStream) (DistributionDownload, error) {
+	info, err := dl.GetFCOSDownload(imageStream)
+	if err != nil {
+		return nil, err
+	}
+	urlSplit := strings.Split(info.Location, "/")
+	dl.ImageName = urlSplit[len(urlSplit)-1]
+	downloadURL, err := url.Parse(info.Location)
+	if err != nil {
+		return nil, err
+	}
+
+	// Complete the download struct
+	dl.Arch = GetFcosArch()
```

Comment:
To be sure: are any of these assignments expected to be permanent and extend the scope of the call?

If the assignments are to be made permanent, the receiver `(dl Download)` must be changed to a pointer receiver `(dl *Download)`.

Reply:
I *think* it is OK as is.